### ISSUE TITLE
Reinstall nuked my existing KBs!!!

### ISSUE THREAD

**Issue by dnorthup-ums** [gnebbia/kb#107]
## Expected Behavior
Re-install to cope with OS upgrade, be able to use my existing KBs

## Actual Behavior
It DELETED THEM.

## Steps to Reproduce the Problem

  1. Upgrade OS, therefore upgrading the system python
  1. Find that KB doesn't work anymore
  1. Re-install KB via pip
  2. Find that KB re-install removed completely the existing store of KBs

## Specifications

(Note, this is kinda useless in this case, but whatever)

  - Version: NEW is latest, today (2023-05-16), from pip
    (pre-upgrade may have been 0.1.6, current seems to be 0.1.7; I don't trust it anymore)
  - Platform: Was F36, Now F37
  - Subsystem: ????

**Comment by dnorthup-ums** [comment]
Note, it is going to take me at least 8 hours to regenerate this information, at best, so I'm very angry right now.

**Comment by dnorthup-ums** [comment]
I found the answer, and I'm still not pleased. If a package is going to completely abandon the previous data storage location (in 0.1.6 and earlier: `~/.kb`) then upon upgrade it must check the previous location for data and migrate it to the new location instead of creating a new data store from scratch. If I find the neurons to do the python-fu I'll let you know. More than anything else, I'm leaving this here in case somebody else has the same problem I did.
